models/stock_lot.py

```diff
@@ -0,0 +1,19 @@
+from odoo import fields, models, api
+from datetime import datetime, timedelta
+
+class StockLot(models.Model):
+    _inherit = 'stock.lot'
+
+    flower_water_ids = fields.One2many('flower.water', 'serial_id')
+    is_flower = fields.Boolean(related='product_id.is_flower')
+    flower_id = fields.Many2one(related='product_id.flower_id')
+
+    def action_water_flower(self):
+        for record in self.filtered(lambda x: x.is_flower and x.flower_id):
+                watering_frequency = record.flower_id.watering_frequency
+                if not record.flower_water_ids or \
+                   record.flower_water_ids[-1].water_datetime + timedelta(days=watering_frequency) <= datetime.now():
```

Comment:
This check could also be used as a constraint during the creation of the flower.water record. Alternatively, you could raise an exception if the if condition does not evaluate to True

```suggestion
if not record.flower_water_ids or \
                   record.flower_water_ids[-1].water_datetime + timedelta(days=watering_frequency) <= datetime.now():
else:
    raise UserError(_("Cannot water the flower currently. Please try again tomorrow."))
``